### PR TITLE
Choice rule payload validation path

### PR DESCRIPTION
**Pull request by kbrock**
extracted from #189 

# Before

If the path referenced a missing value in the input, treat it as a nil.

# After

If the path references a missing value in the input, raise an error.

- introduced `presence_check` method to check if it is present up front.
- This also affects Intrinsic methods. You can see the changed spec.
- This respects `IsPresent: false` and `IsPresent: true`

### CODE REVIEW

File: spec/workflow/path_spec.rb [hunk 1]
```diff
@@ -11,15 +11,15 @@
   describe "#value" do
     context "referencing the global context" do
       it "with a missing value" do
-        expect(described_class.new("$$.foo").value({})).to be_nil
+        expect { described_class.new("$$.foo").value({}, {"foo" => "bar"}) }.to raise_error(Floe::PathError, "Path [$$.foo] references an invalid value")
```
Comment (kbrock) [inline spec/workflow/path_spec.rb:14]: this is the essence of the issue

**Comment by kbrock** [comment]
update:
- removed spec updates/refactor (I'll put in another PR)

**Comment by kbrock** [comment]
update:
- removed changes to `is_present?` -- it is not useful now, will delete after the dust settles.

**Comment by kbrock** [comment]
WIP: moving stuff around. Want to get other stuff merged before this.
There is no outstanding work for this item.

**Comment by kbrock** [comment]
WIP: oops. thought this was something else.
This is good to go

**Review by agrare — APPROVED** [review]
:+1: LGTM

**Pull request merged**